File: bench/fibo.pl
use v5;
use strict;

sub fibo {
    my $n = shift;
    if ($n == 0) {
        return 0;
    }
    if ($n == 1) {
        return 1;
    }
    return fibo($n - 1) + fibo($n - 2);
}

my $i = 0;
for ($i = 0; $i < 30; $i++) {
    print(fibo($i), "\n");
}

File: bench/fibo.go
package main

import (
    "fmt"
)

func fibo(n int) int {
    if n == 0 {
        return 0
    } else if n == 1 {
        return 1
    }
    return fibo(n - 1) + fibo(n - 2)
}

func main() {
    for i := 0; i < 30; i++ {
        fmt.Printf("%d\n", fibo(i))
    }
}
